source/kernel/src/daemons/shells.rs

```diff
@@ -19,43 +27,150 @@ use profont::PROFONT_12_POINT;
 
 use crate::forth::Forth;
 
-// ----
+/// Settings for the [sermux_shell] daemon
+#[derive(Debug)]
+pub struct SermuxShellSettings {
+    /// Sermux port to serve the shell on
+    ///
+    /// Defaults to [WellKnown::ForthShell0]
+    pub port: u16,
+    /// Number of bytes used for the sermux buffer
+    ///
+    /// Defaults to 256
+    pub capacity: usize,
+    /// Forth parameters for the shell
+    ///
+    /// Uses the default value of [Params]
+    pub forth_settings: Params,
+    /// Hidden for forwards compat
+    _priv: (),
+}
 
+impl Default for SermuxShellSettings {
+    fn default() -> Self {
+        Self {
+            port: WellKnown::ForthShell0.into(),
+            capacity: 256,
+            forth_settings: Default::default(),
+            _priv: (),
+        }
+    }
+}
+
+/// Spawns a forth shell on the given port
 #[tracing::instrument(skip(k))]
-pub async fn graphical_shell_mono(
-    k: &'static Kernel,
-    disp_width_px: u32,
-    disp_height_px: u32,
-    port: u16,
-    capacity: usize,
-) {
+pub async fn sermux_shell(k: &'static Kernel, settings: SermuxShellSettings) {
+    let SermuxShellSettings {
+        port,
+        capacity,
+        forth_settings,
+        _priv,
+    } = settings;
+    let port = PortHandle::open(k, port, capacity).await.unwrap();
+    let (task, tid_io) = Forth::new(k, forth_settings)
+        .await
+        .expect("Forth spawning must succeed");
+    k.spawn(task.run()).await;
+    k.spawn(async move {
+        loop {
+            futures::select_biased! {
+                rgr = port.consumer().read_grant().fuse() => {
+                    let needed = rgr.len();
+                    let mut tid_io_wgr = tid_io.producer().send_grant_exact(needed).await;
+                    tid_io_wgr.copy_from_slice(&rgr);
+                    tid_io_wgr.commit(needed);
+                    rgr.release(needed);
+                },
+                output = tid_io.consumer().read_grant().fuse() => {
+                    let needed = output.len();
+                    port.send(&output).await;
+                    output.release(needed);
+                }
+            }
+        }
+    })
+    .await;
+}
+
+/// Settings for the [graphical_shell_mono] daemon
+///
+/// This does NOT implement [Default]. Instead use [GraphicalShellSettings::with_display_size].
```

Comment:
maybe worth showing an example of this initializer:
```suggestion
/// This does NOT implement [Default]. Instead use [GraphicalShellSettings::with_display_size].
///
/// For example:
/// ```
/// use kernel::daemons::shells::GraphicalShellSettings;
/// let shell = GraphicalShellSettings {
///     // override the capacity with a larger value:
///     capacity: 512,
///    ..GraphicalShellSettings::with_display_size(420, 69), // nice!
/// };
/// # drop(shell);
/// ```
```